modules/run-pex-as-resource/main.tf

```diff
@@ -18,7 +18,24 @@ module "pex_env" {
 resource "null_resource" "run_pex" {
   count = var.enabled ? 1 : 0
 
-  triggers = var.triggers
+  triggers = merge(
+    var.triggers != null ? var.triggers : {},
+
+    # We store various values in the triggers so that they can be used in the destroy time provisioner.
+    # See https://github.com/hashicorp/terraform/issues/23679 for more details.
+    {
+      __pythonpath = module.pex_env.python_path
+      __env        = jsonencode(var.env)
+      __destroy_provisioner_command_args = (
+        var.enable_destroy_provisioner
+        # NOTE: The nested string interpolation can not be extracted because of the reference to self.
+        ? "${local.python_call} ${var.destroy_command_args}"
+        : "echo 'Skipping delete provisioner'"
+      )
+      __destroy_provisioner_pass_in_previous_triggers = var.pass_in_previous_triggers ? "true" : "false"
+      __destroy_provisioner_previous_trigger_option   = var.previous_trigger_option
```

Comment:
Won't this cause the `null_resource` to be recreated any time any of those params changes? In other words, wouldn't this have the side effect of running create _and_ destroy far more frequently than we might otherwise expect?